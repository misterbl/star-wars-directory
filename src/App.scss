$dark-grey-color: #282727;

$pink-color: #9e4f60;

%grey-input {
  input,
  input:focus,
  input:active {
    outline: none;
    background-color: $dark-grey-color;
    border: none;
    font-size: 30px;
    height: 60px;
  }
}
%avatar-container {
  height: 90px;
  width: 90px;
  border-radius: 15%;
  min-width: 95px;
}
html {
  height: 100%;
}
body {
  height: 100%;
  margin: 0px 0px 50px 0px;
  padding: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  background-image: url("./assets/star-background.png");
  background-position: 0px -300px;
  background-size: contain;
}

header {
  background-image: url("./assets/star-wars-header-small.png");
  height: 93px;
  position: relative;
  background-position: 0;
  background-size: contain;
  background-repeat: no-repeat;
  background-color: #222222;
  @media only screen and (min-width: 1024px) {
    height: 250px;
    background-size: cover;
    background-color: transparent;
    background-position: initial;
    background-image: url("./assets/star-wars-header.png");
  }
}

.form-control::-webkit-inner-spin-button,
.form-control::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.fatal-error {
  background-image: url("./assets/error-page.jpg");
  background-size: cover;
  height: 110%;
  .back-button {
    left: 15px;
  }
}

#root {
  height: 100%;
}

.avatar {
  &__default {
    @extend %avatar-container;
    background-image: url("./assets//star-wars_characters.jpg");
    background-size: cover;
  }
  &__film {
    background-image: url("./assets/star_wars.png");
    background-size: cover;
    width: 190px;
  }
  div {
    background-image: url("./assets/star-war-characters-grid.jpg");
    @extend %avatar-container;
  }
}

.search-form {
  &__header {
    display: none;
    @media only screen and (min-width: 1024px) {
      display: block;
      width: 300px;
      position: absolute;
      right: 20px;
      border-radius: 4px;
      button {
        background-image: url("./assets/search.svg");
        width: 25px;
        height: 25px;
        position: absolute;
        right: 14px;
        top: 30px;
        background-size: cover;
        border: none;
      }
    }
  }
  &__body {
    @extend %grey-input;
    position: relative;
    border-radius: 15px;
    @media only screen and (min-width: 1024px) {
      display: none;
    }
    button {
      background-image: url("./assets/search-button.png");
      width: 120px;
      height: 41px;
      position: absolute;
      right: 12px;
      top: 34px;
      background-size: cover;
      border: none;
      border-radius: 5px;
    }
  }
}

.btn-dark {
  background-color: $dark-grey-color;
}

.sms-form {
  @extend %grey-input;
  border-radius: 15px;
}

.back-button {
  background-image: url("./assets/leftArrow.svg");
  width: 25px;
  height: 25px;
  position: absolute;
  left: 63px;
  background-size: cover;
  border: none;
}

.resistance {
  background-image: url("./assets/resistance.png");
  height: 110px;
  background-size: cover;
  border-radius: 10px;
  cursor: pointer;
}

.grey-line {
  border-bottom: 1px solid $dark-grey-color;
}

.info-details {
  border-left: 2px solid $pink-color;
  background-color: $dark-grey-color;
  color: white;
  opacity: 0.7;
  p {
    margin-bottom: 3px;
    font-size: 18px;
  }
}

.film-image {
  width: 400px;
  height: auto;
}

.pointer {
  cursor: pointer;
}

.loader {
  position: absolute;
  top: 40%;
  left: 40%;
  img {
    width: 300px;
  }
}

.result-filter {
  background-color: $dark-grey-color;
  color: white;
  opacity: 0.7;
  display: flex;
  justify-content: space-between;
}

.custom-select {
  background-color: $dark-grey-color;
  border: none;
  color: white;
}

.info-category {
  border-left: 2px solid $dark-grey-color;
  padding: 0px 15px 0px 15px;
  margin-top: 15px;
  div {
    font-size: 15px;
    color: white;
    margin-bottom: 8px;
  }
  p {
    color: $pink-color;
    font-size: 14px;
  }
}

.film-card {
  width: 480px;
}
