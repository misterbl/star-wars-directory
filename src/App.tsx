import React, { Component } from "react";
import { Route, Switch, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch } from "redux";
import ROUTES from "./const/routes";
import Header from "./components/Header";
import Home from "./containers/Home";
import CharacterInfo from "./containers/CharacterInfo";
import FilmInfo from "./containers/FilmInfo";
import { IAppComponent } from "./App.d";
import { isLoading, isFatalError } from "./selectors/appSelectors";
import { IAppState } from "./state";
import FatalError from "./components/FatalError";
import { setFatalError } from "./actions/actionCreators/actions";
import lightSaber from "./assets/lightsaber.gif";
export class App extends Component<IAppComponent> {
  resetFatalError = () => {
    this.props.setFatalError(false);
  };
  render() {
    const {
      isLoading,
      isFatalError,
      history: { push }
    } = this.props;
    if (isFatalError === true) {
      return <FatalError push={push} resetFatalError={this.resetFatalError} />;
    } else if (isLoading) {
      return (
        <div className="loader">
          <img src={lightSaber} alt="loading..." />
        </div>
      );
    }
    return (
      <>
        <Header />
        <Switch>
          <Route exact path={ROUTES.INDEX} component={Home} />
          <Route path={ROUTES.CHARACTER} component={CharacterInfo} />
          <Route path={ROUTES.FILM} component={FilmInfo} />
        </Switch>
      </>
    );
  }
}

export const mapStateToProps = (state: IAppState) => ({
  isLoading: isLoading(state),
  isFatalError: isFatalError(state)
});

export const mapDispatchToProps = (dispatch: Dispatch) => ({
  setFatalError: bindActionCreators(setFatalError, dispatch)
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(App)
);
