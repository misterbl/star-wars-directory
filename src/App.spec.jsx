/* eslint-disable no-undef */
import * as React from "react";
import { shallow } from "enzyme";
import history from "./testMocks/history.mock";
import { App } from "./App";

describe("App", () => {
  const props = {
    isLoading: false,
    isFatalError: false,
    history
  };
  const wrapper = shallow(<App {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("shows the loading indicator if isLoading is true", () => {
    wrapper.setProps({ isLoading: true });
    expect(wrapper.find(".loader").length).toEqual(1);
  });
  it("shows the FatalError page if isFatalError is true", () => {
    wrapper.setProps({ isFatalError: true });
    expect(wrapper.find("FatalError").length).toEqual(1);
  });
});
