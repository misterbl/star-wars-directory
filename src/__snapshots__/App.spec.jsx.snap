// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    history={
      Object {
        "action": "PUSH",
        "block": [MockFunction],
        "createHref": [MockFunction],
        "go": [MockFunction],
        "goBack": [MockFunction],
        "goForward": [MockFunction],
        "length": 123,
        "listen": [MockFunction],
        "location": Object {
          "hash": "",
          "key": "",
          "pathname": "",
          "search": "",
          "state": "",
        },
        "push": [MockFunction],
        "replace": [MockFunction],
      }
    }
    isFatalError={false}
    isLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <withRouter(Connect(Header)) />,
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/character"
          />
          <Route
            component={[Function]}
            path="/film"
          />
        </Switch>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              path="/character"
            />,
            <Route
              component={[Function]}
              path="/film"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/character",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/film",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <withRouter(Connect(Header)) />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/character"
            />
            <Route
              component={[Function]}
              path="/film"
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                path="/character"
              />,
              <Route
                component={[Function]}
                path="/film"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/character",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/film",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
